.github/workflows/benchmarks.yml

```
@@ -27,8 +27,12 @@ jobs:
             ~/.cargo/registry/cache/
             ~/.cargo/git/db/
             target/
-          key: ${{ runner.os }}-cargo-${{ hashFiles('Cargo.lock') }}
+          key:
+            ${{ runner.os }}-cargo-${{ hashFiles('Cargo.lock') }}-${{
+            github.run_id }}
           fail-on-cache-miss: true
+          restore-keys: |
+            ${{ runner.os }}-cargo
```

exact match (we want to execute on the build of the run)

---

Do I understand correctly that you'd like multiple restore keys?

```suggestion
     ${{ runner.os }}-cargo-${{ hashFiles('Cargo.lock') }}-${{
            github.run_id }}
     ${{ runner.os }}-cargo-${{ hashFiles('Cargo.lock') }}
     ${{ runner.os }}-cargo
```

---

on the build yes, on the retrieve no